cpp/ql/src/experimental/Security/CWE/CWE-359/PrivateCleartextWrite.ql

```diff
@@ -0,0 +1,18 @@
+/**
+ * @name Exposure of private information
+ * @description If private information is written to an external location, it may be accessible by
+ *              unauthorized persons.
+ * @kind path-problem
+ * @problem.severity error
+ * @id cpp/private-cleartext-write
+ * @tags security
+ *       external/cwe/cwe-359
+ */
+
+import cpp
+import experimental.semmle.code.cpp.security.PrivateCleartextWrite
+import experimental.semmle.code.cpp.security.PrivateCleartextWrite::PrivateCleartextWrite
+
+from WriteConfig b, DataFlow::Node source, DataFlow::Node sink
+where b.hasFlow(source, sink)
+select sink, "This write may contain unencrypted data"
```

Comment:
This doesn't work with `@kind path-problem`, since it needs a slightly different pattern of results. That's causing the language-tests failure. See https://help.semmle.com/QL/learn-ql/writing-queries/path-queries.html#constructing-a-path-query for the pattern you need.

We also generally include the source with a `$@` placeholder to make results a little more friendly to end users.